Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions
+        : HttpResponseMessageAssertions<HttpResponseMessageAssertions>
+    {
+        /// <summary>Initializes a new instance of the <see cref="HttpResponseMessageAssertions"/> class.</summary>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="value"/> is <c>null</c></exception>
+        public HttpResponseMessageAssertions(HttpResponseMessage value)
+            : base(value)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions<TAssertions>
+        where TAssertions : HttpResponseMessageAssertions<TAssertions>
+    {
+        /// <summary>Initializes a new instance of the <see cref="HttpResponseMessageAssertions{TAssertions}"/> class.</summary>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="value"/> is <c>null</c></exception>
+        public HttpResponseMessageAssertions(HttpResponseMessage value)
+        {
+            Guard.ThrowIfArgumentIsNull(value, nameof(value));
```

Comment:
❌ We still need to properly document the [design guidelines](https://github.com/fluentassertions/fluentassertions/issues/1340), but we use guards to capture misuse of our library. However, a `null` message is something that FA should be prepared for. In those cases, we use a proper `FailWith` construct, preferably per assertion method.